test-suite/Makefile

```diff
@@ -242,6 +251,52 @@ $(addsuffix .log,$(wildcard bugs/closed/*.v)): %.v.log: %.v
 	  fi; \
 	} > "$@"
 
+#######################################################################
+# Unit tests
+#######################################################################
+
+OCAMLOPT := $(OCAMLFIND) opt $(CAMLFLAGS)
+SYSMOD:=-package num,str,unix,dynlink,threads
+
+# why intf? not listed in CORESRCDIRS in Makefile.common, no dir in source tree but exists in installed
+UNIT_COQDIRS=-I $(LIB)/clib -I $(LIB)/lib -I $(LIB)/kernel -I $(LIB)/kernel/byterun -I $(LIB)/intf \
+-I $(LIB)/library -I $(LIB)/engine -I $(LIB)/pretyping -I $(LIB)/interp \
+-I $(LIB)/proofs -I $(LIB)/parsing -I $(LIB)/printing -I $(LIB)/tactics -I $(LIB)/vernac \
+-I $(LIB)/stm -I $(LIB)/toplevel
+
+# ML files from unit-test framework, not containing tests
+UNIT_SRCFILES=$(shell find ./unit-tests/src -name *.ml)
+UNIT_ALLMLFILES:=$(shell find ./unit-tests -name *.ml)
+UNIT_MLFILES:=$(filter-out $(UNIT_SRCFILES),$(UNIT_ALLMLFILES))
+UNIT_LOGFILES:=$(patsubst %.ml,%.ml.log,$(UNIT_MLFILES))
+
+UNIT_CMXS=utest.cmx
+
+# this is same link order as CORECMA in Makefile.common in Coq root dir
+UNIT_COQCMXAS=clib.cmxa lib.cmxa kernel.cmxa library.cmxa engine.cmxa pretyping.cmxa \
+ interp.cmxa proofs.cmxa parsing.cmxa printing.cmxa tactics.cmxa vernac.cmxa stm.cmxa toplevel.cmxa
+
+unit-tests/src/utest.cmx : unit-tests/src/utest.ml unit-tests/src/utest.cmi
+	$(SHOW)'OCAMLOPT  $<'
+	$(HIDE)$(OCAMLOPT) -c -I unit-tests/src -package oUnit unit-tests/src/utest.ml
+unit-tests/src/utest.cmi : unit-tests/src/utest.mli
+	$(SHOW)'OCAMLOPT  $<'
+	$(HIDE)$(OCAMLOPT) -package oUnit unit-tests/src/utest.mli
+
+$(UNIT_LOGFILES) : unit-tests/src/utest.cmx
+
+unit-tests: $(UNIT_LOGFILES)
+
+# Build executable, run it to generate log file
+%.ml.log : %.ml
+	$(SHOW)'TEST      $<'
+	$(HIDE)$(OCAMLOPT) -linkall -linkpkg -I vernac -I toplevel \
+	                   -I kernel/byterun/ -cclib -lcoqrun \
```

Comment:
What's going on here?

Reply:
It was needed to get the unit tests in my other PR to compile and run.  I compared the parameters used for `make coqbinaries` and gradually added things that were needed.  A whole series of experiments.

```
> make coqbinaries
"C:\\OCaml64\\home\\Jim\\.opam\\4.02.3+mingw64c\\bin/ocamlfind.exe" opt -thread -rectypes -w +a-4-9-27-41-42-44-45-48-50   -safe-string  -linkall -linkpkg -I vernac -I toplevel \
                   -I kernel/byterun/ -cclib -lcoqrun \
                   -package num,str,unix,dynlink,threads -package camlp5.gramlib \
                   clib/clib.cmxa lib/lib.cmxa kernel/kernel.cmxa library/library.cmxa engine/engine.cmxa pretyping/pretyping.cmxa interp/interp.cmxa proofs/proofs.cmxa parsing/parsing.cmxa printing/printing.cmxa tactics/tactics.cmxa vernac/vernac.cmxa stm/stm.cmxa toplevel/toplevel.cmxa  -g     toplevel/coqtop_opt_bin.ml -o bin/coqtop.exe
```